src/foundry/foundry.js/clientDocuments/combatant.d.ts

```diff
@@ -0,0 +1,101 @@
+import { ConfiguredDocumentClass } from '../../../types/helperTypes';
+import { BaseUser } from '../../common/documents.mjs';
+
+declare global {
+  /**
+   * The client-side Combatant document which extends the common BaseCombatant model.
+   * Each Combatant belongs to the effects collection of its parent Document.
+   * Each Combatant contains a CombatantData object which provides its source data.
+   *
+   * @see {@link data.CombatantData}                  The Combatant data schema
+   * @see {@link documents.Combat}                    The Combat document which contains Combatant embedded documents
+   * @see {@link applications.CombatantConfig}        The Combatant configuration application
+   */
+  class Combatant extends ClientDocumentMixin(foundry.documents.BaseCombatant) {
+    /**
+     * @override
+     * @param data    - Initial data provided to construct the Combatant document
+     * @param context - The document context, see {@link foundry.abstract.Document}
+     */
+    constructor(
+      data: ConstructorParameters<typeof foundry.documents.BaseCombatant>[0],
+      context: ConstructorParameters<typeof foundry.documents.BaseCombatant>[1]
+    );
+
+    /** A cached reference to the Token which this Combatant represents, if any */
+    protected _token: InstanceType<ConfiguredDocumentClass<typeof foundry.documents.BaseToken>> | null;
+
+    /** A cached reference to the Actor which this Combatant represents, if any */
+    protected _actor: InstanceType<ConfiguredDocumentClass<typeof foundry.documents.BaseActor>> | null;
+
+    /** The current value of the special tracked resource which pertains to this Combatant */
+    resource: `${number}` | number | boolean | null;
```

Comment:
consider creating an issue at foundry, the documented type of `     * @type {object|null}` is incorrect.

Reply:
will make an issue